robottelo/config/virtwho.py

```diff
@@ -0,0 +1,499 @@
+                '[rhevm] hypervisor_type, hypervisor_server, hypervisor_username, '
+                'hypervisor_password, guest , guest_username, guest_password, '
+                'sku_vdc_physical, sku_vdc_virtual options must be provided.'
+            )
+        return validation_errors
+
+
+class LibvirtSettings(FeatureSettings):
+    """VirtWho settings definitions."""
+
+    def __init__(self, *args, **kwargs):
+        super(LibvirtSettings, self).__init__(*args, **kwargs)
+        # Hypervisor Information
+        self.hypervisor_type = None
+        self.hypervisor_server = None
+        self.hypervisor_username = None
+        self.hypervisor_password = None
+        self.guest = None
+        self.guest_port = None
+        self.guest_username = None
+        self.guest_password = None
+        # SKU Information
+        self.sku_vdc_physical = None
+        self.sku_vdc_virtual = None
+
+    def read(self, reader):
+        """Read virtwho settings."""
+        # Hypervisor Information
+        self.hypervisor_type = reader.get('libvirt', 'hypervisor_type')
+        self.hypervisor_server = reader.get('libvirt', 'hypervisor_server')
+        self.hypervisor_username = reader.get('libvirt', 'hypervisor_username')
+        self.hypervisor_password = reader.get('libvirt', 'hypervisor_password')
+        self.guest = reader.get('libvirt', 'guest')
+        self.guest_port = reader.get('libvirt', 'guest_port', 22, int)
+        self.guest_username = reader.get('libvirt', 'guest_username')
+        self.guest_password = reader.get('libvirt', 'guest_password')
+        # SKU Information
+        self.sku_vdc_physical = reader.get('libvirt', 'sku_vdc_physical')
+        self.sku_vdc_virtual = reader.get('libvirt', 'sku_vdc_virtual')
+
+    def validate(self):
+        """Validate virtwho settings."""
+        validation_errors = []
+        mandatory = (
+            self.hypervisor_type,
+            self.hypervisor_server,
+            self.hypervisor_username,
+            self.hypervisor_password,
+            self.guest,
+            self.guest_username,
+            self.guest_password,
+            self.sku_vdc_physical,
+            self.sku_vdc_virtual,
+        )
+        if not all(mandatory):
+            validation_errors.append(
+                '[libvirt] hypervisor_type, hypervisor_server, hypervisor_username, '
+                'hypervisor_password, guest , guest_username, guest_password, '
+                'sku_vdc_physical, sku_vdc_virtual options must be provided.'
+            )
+        return validation_errors
+
+
+class KubevirtSettings(FeatureSettings):
+    """VirtWho settings definitions."""
+
+    def __init__(self, *args, **kwargs):
+        super(KubevirtSettings, self).__init__(*args, **kwargs)
+        # Hypervisor Information
+        self.hypervisor_type = None
+        self.hypervisor_server = None
+        self.hypervisor_username = None
+        self.hypervisor_password = None
```

Comment:
username and password are not available for kubevirt mode

Reply:
Also deleted them